test/unit/local/steps/initial_diff.js

```diff
@@ -0,0 +1,100 @@
+/* eslint-env mocha */
+/* @flow */
+
+const should = require('should')
+const Builders = require('../../../support/builders')
+const configHelpers = require('../../../support/helpers/config')
+const pouchHelpers = require('../../../support/helpers/pouch')
+
+const Buffer = require('../../../../core/local/steps/buffer')
+const initialDiff = require('../../../../core/local/steps/initial_diff')
+
+describe('local/steps/initialDiff', () => {
+  let builders
+  let buffer
+
+  before('instanciate config', configHelpers.createConfig)
+  beforeEach('instanciate pouch', pouchHelpers.createDatabase)
+  beforeEach('populate pouch with documents', function () {
+    builders = new Builders({pouch: this.pouch})
+    buffer = new Buffer()
+  })
+  afterEach('clean pouch', pouchHelpers.cleanDatabase)
+  after('clean config directory', configHelpers.cleanConfig)
+
+  it('detects documents moved while client was stopped', async function () {
+    builders.metadir().path('foo').ino(1).create()
+    builders.metafile().path('baz').ino(2).create()
+
+    const bar = builders.event().action('scan').kind('directory').path('bar').stats({ ino: 1 }).build()
+    const buz = builders.event().action('scan').kind('file').path('buz').stats({ ino: 2 }).build()
+    buffer.push([bar, buz])
+    buffer = initialDiff(buffer, { pouch: this.pouch })
+
+    const events = await buffer.pop()
+    should(events).deepEqual([
+      builders.event(bar).action('renamed').oldPath('foo').build(),
+      builders.event(buz).action('renamed').oldPath('baz').build()
```

Comment:
nitpicking: Although reuse makes sense for docs since we're often building the next version based on the previous one, I wonder whether it makes sense for an event to build upon another one...

Reply:
I've done it in some cases where I was testing that an event is replaced by another one with very few attributes changing.
But I haven't done so often.

Reply:
Not a blocker.